pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go

```diff
@@ -231,11 +233,22 @@ type ClusterIntegrations struct {
 
 // Validate ensures there is no conflicting options
 func (c ClusterIntegrations) Validate() error {
-	if c.ArgoCD && !c.Cognito {
-		return errors.New("the identity provider cognito is required when using ArgoCD")
-	}
+	return validation.ValidateStruct(&c,
+		validation.Field(&c.Cognito, validation.When(c.ArgoCD, validation.Required.Error("is required when argocd is enabled"))),
```

Comment:
That looks cleaner than what I did :$

Reply:
The only reason I changed it is because we can defer validation down the structs with `ozzo-validation`: https://github.com/go-ozzo/ozzo-validation#validatable-types, not that this is required here, but anyone reading the code will see that we always use the same pattern, a minor detail I know!